src/containers/board-container.js

```diff
@@ -0,0 +1,118 @@
+import React from 'react';
+import { Light } from '../components/light.js';
+import { Board } from '../components/board.js';
+
+export class BoardContainer extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            board: [],
+            numMoves: 0,
+            winComponent: <p />,
+        };
+        this.board = this.getNewBoard();
+    }
```

Comment:
Now that JS has class properties, you can actually set these initial values without a constructor. I think this is a little nicer, and allows you to move `getNewBoard` out of the class and into a separate function/file, but I also see the value in making it a member of the class. @solomonhawk thoughts?

```js
// get-new-board.js
export default const getNewBoard = () => {
    let board = [];
    let isALightOn = false;
    for (let row = 0; row < 5; row++) {
        board.push([]);
        for (let col = 0; col < 5; col++) {
            let isOn = Math.random() < 0.1 ? true : false;
            if (isOn) isALightOn = true;
            board[row].push(isOn);
        }
    }
    if (isALightOn) {
        return board;
    } else {
        return getNewBoard();
    }
};

// board-container.js

import getNewBoard from './get-new-board'
export class BoardContainer extends React.Component {
    state = {
        board: [],
        numMoves: 0,
        winComponent: <p />,
    }

    board = getNewBoard()
```